analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs

```diff
@@ -215,24 +216,29 @@ variableDeclarator.Parent.Parent switch
                 };
 
             Diagnostic CreateS1144Diagnostic(SyntaxNode syntaxNode, ISymbol symbol) =>
-                Diagnostic.Create(RuleS1144, syntaxNode.GetLocation(), accessibility, symbol.GetClassification(), GetMemberName(symbol));
+                Diagnostic.Create(RuleS1144, GetIssueLocation(syntaxNode), accessibility, symbol.GetClassification(), GetMemberName(symbol));
```

Comment:
We need to discuss what to do about the `fadeOutCode` argument of the diagnostic. The best solution would be to fade out the entire method (current behavior) while squiggling only the method name at the same time. I don't think this is possible but you may be able to find a good workaround.

Reply:
can you bring more detail on this, @martin-strecker-sonarsource ? 

I looked in the (both the version we use and the [latest](https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs)) and could not find this argument

Reply:
This is what it looks like for me for unused methods in VS:
![image](https://user-images.githubusercontent.com/103252490/215980132-0480ed68-477a-4791-8039-6074bfe6abec.png)

The `fadeOutCode` parameter dims the whole method which looks nice (see e.g. the difference between `private` and `protected`. But at the same time, all the squiggles are much too offensive.
So for me fading out the whole method while only squiggling the method (or don't squiggle at all) would be the best compromise.

If we only squiggle e.g. the method name I would remove the `fadeOutCode` option because that would most likely look awkward.

Reply:
Thank you, that's good to know 😃 I actually thought we were doing the squiggly and VS is doing the fading out by itself.

I checked the code, `fadeoutCode `is translated to `WellKnownDiagnosticTags.Unnecessary` and it applies to the entire Diagnostic. I don't see any way of applying it partially. The only way I see to do this is to raise a secondary issue, which only does the fading of the code. But that sounds like an awkward solution.
What do think?

Reply:
If you want to go crazy, you can do what Roslyn does in such a case:
https://github.com/dotnet/roslyn/blob/1d8f8cd8be6581a03e8379b8f146dcb5f11800c9/src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs#L64-L105

See how `additionalUnnecessaryLocations` is handled. But you may ask @andrei-epure-sonarsource if he sees the need for us to support a scenario where the fadeout is different from the squiggles.

Reply:
this is nice and interesting to see @martin-strecker-sonarsource !

I would favor to keep this PR simple - adding this new functionality to the analyzer should come with a change in the testing framework too, and we'd go too wide for this issue

I think it would be a cool thing, but I would also check where else this might help and do the updates to all the rules - this should probably be an MMF on improving visibility and useability of issues inside IDEs

Reply:
I created #6697

Reply:
Thank you. I disabled the fade-out effect for now for both rules.